src/accelerate/utils/dataclasses.py

```diff
@@ -529,6 +530,23 @@ def set_deepspeed_weakref(self):
 
             self.dschf = HfDeepSpeedConfig(ds_config)  # keep this object alive # noqa
 
+    def is_zero3_init_enabled(self):
+        return self.zero3_init_flag
+
+    @contextmanager
+    def set_zero3_init(self, enable=False):
+        old = self.zero3_init_flag
+        if old == enable:
+            yield
+            return
+        self.zero3_init_flag = enable
+        self.dschf = None
+        self.set_deepspeed_weakref()
+        yield
+        self.zero3_init_flag = old
+        self.dschf = None
+        self.set_deepspeed_weakref()
```

Comment:
Might be clearer to have an `else` block here instead of the early `return`.

Reply:
Done.